examples/data_device.rs

```diff
@@ -652,6 +653,8 @@ impl DataDeviceHandler for DataDeviceWindow {
                     (o, d, Some(t)) => (o, d, t),
                     _ => return,
                 };
+                // SAFETY: it's safe as long as we don't close the underlying file.
+                let f: &mut fs::File = unsafe { f.get_mut() };
```

Comment:
This is annoying, though I'm not sure of a better way to do it here or in calloop while preserving IO safety. (And also not over complicating things; like manual syscalls with rustix instead of using `File` methods).

Reply:
Yes, I've discussed this in #calloop channel, basically the issue is that `NoIoDrop` thing can't be re-wrapped by us...

Reply:
Oh, with the alternative of doing a `dup`, but there's no way I'll do a `dup` for that.

Reply:
`dup` works, but yeah, not great. Technically also perfectly safe to define a struct implementing `Read` using `rustix::io::read` on the fd... but that would be rather verbose if nothing else.

Reply:
Actually, I'm thinking in the case of SCTK, handling pipes from data device would be a place where using the futures/IO adapters from calloop could make quite a bit of sense :thinking:

Reply:
I'm not sure about Futures, but yeah, if we write a specialized object which simply read/writes maybe it'll be nicer in the end, I just ported the API.